### ISSUE TITLE
[Crash] bad argument #7 to 'fmt' (string expected, got nil)

### ISSUE THREAD

**Issue by h4ml3t** [minetest-mods/craftguide#88]
Craftguide v1.8; MT v5.0.1; MTG v5.0.1; OS Debian 10

This happens with the binoculars:

ERROR[Main]: ServerError: AsyncErr: ServerThread::run Lua: Runtime error from mod 'craftguide' in callback on_playerReceiveFields(): .../craftguide/init.lua:621: bad argument #7 to 'fmt' (string expected, got nil)
ERROR[Main]: stack traceback:
ERROR[Main]: 	[C]: in function 'fmt'
ERROR[Main]: 	.../craftguide/init.lua:621: in function 'get_recipe_fs'
ERROR[Main]: 	.../craftguide/init.lua:724: in function 'make_formspec'
ERROR[Main]: 	.../craftguide/init.lua:734: in function 'show_fs'
ERROR[Main]: 	.../craftguide/init.lua:1008: in function 'fields'
ERROR[Main]: 	.../craftguide/init.lua:1041: in function '?'
ERROR[Main]: 	.../minetest/bin/../builtin/game/register.lua:419: in function <.../minetest/bin/../builtin/game/register.lua:399>

**Comment by kilbith** [comment]
Cannot reproduce. Please update the mod to the lastest commit and retry.

**Comment by h4ml3t** [comment]
Done, same as above:

Screenshot 1 https://imgur.com/PcDGTpW
Screenshot 2 https://imgur.com/adoDTrB
Gallery: https://imgur.com/a/E7QPXkQ

**Comment by kilbith** [comment]
Well, that doesn't make any sense, it shouldn't be `nil` at all. Anyway, retry with the last commit: https://github.com/minetest-mods/craftguide/commit/7a8b8d19df7c7e38434b18560734b5d2bc1ea84b

**Comment by h4ml3t** [comment]
Tried, still having the same issue.
However it doesn't happen on Win7.

If it can be of any help I've checked which recipes produce that error, here is the image gallery (red = error): https://imgur.com/a/DgxIkOJ

I must say that I haven't checked each and every stair (pages 5-7) but I've randomly checked some of them so I think that it is safe to assume that all of them produce that error (assuming that the recipes are registered in the same manner for all of them, of course)... but I would not bet on it: e.g. some rails are safe, some are not.
